packages/core/core/src/services/document-service/document-engine.ts

```diff
@@ -313,7 +313,7 @@ const createDocumentEngine = ({
     return this.delete(uid, documentId, {
       ...params,
       lookup: { ...params?.lookup, publishedAt: { $ne: null } },
-    }) as any;
+    }).then(({ deletedEntries }) => ({ versions: deletedEntries })) as any;
```

Comment:
that makes me think 🤔  unpublishing can never overwrite the draft  before deletion for now ? maybe we will want to allow it later.

Reply:
not directly, and we do have that exact same usecase on the content manager. We are discarding the draft + unpublishing